examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -742,13 +763,43 @@ def _viz_fetcher(self, post_sim_update_dict):
                 )
             )
 
+    def _reset_current_episode(self):
+        # save initial agent states
+        prev_init_agent_states = self._init_agent_states
+        self._init_agent_states = {}
+
+        # reset current episode
+        # env's current_episode setter will set internal attribute
+        # _episode_from_iter_on_reset to False as a result reset
+        # will be called without the episode changing
+        self._sandbox_service.env.current_episode = self._current_episode
+        self._sandbox_service.end_episode(do_reset=True)
+
+        # restore initial agent states
+        sim = self.get_sim()
```

Comment:
I don't want a custom reset codepath in the HITL codebase. It would be tedious to maintain. We should find a way to re-use habitat-lab episode-resetting (which should correctly initialize articulated objects and everything else in the scene).